app/controllers/movie_preference_controller.py

```diff
@@ -36,9 +36,15 @@ def update(self, user, movie_preference_id):
     def get(self, user):
         user_movie_preferences = user.movies
 
+        # Wrap get_movie calls inside lambdas, then call these lambdas in parallel.
+        get_movie_lambdas = map(lambda mp: (lambda: get_movie(mp.external_movie_id)), user_movie_preferences)
```

Comment:
Ok thought (mostly related to thread stuff having read more):

```
results_by_mp = {}
movie_lambdas = [lambda mp: results_by_mp[mp] = get_movie(mp.external_movie_id) for mp in user_movie_preferences]
results = call_parallel(movie_lambdas)

for mp, result in results_by_mp:
    # Do stuff to create dict
```
Then we get around the weird return issue, and we also don't need to worry about `zip`. The `zip` thing is quasi-superstitious, but there's like risk of bugs later b/c the zip depends on ordering, so if the ordering ever gets fucked up then the code's buggy (and ordering is hard to reason about w/ threads). The way you have it right now I think it's always safe, but it's like a concern in the future.

(also minor, changed `get_movie_lambdas` to `movie_lambdas` since the value is a list, not a function.